syntax="proto3";

package eigenvector;

service EigenVector {
  rpc GetEigenVector (stream EigenVectorRequest) returns (stream EigenVectorResponse);
}

message RealRows {
  repeated double row = 1;
}

message ComplexNumber {
  double real = 1;
  double imaginary = 2;
}

message ComplexRows {
  repeated ComplexNumber row = 1;
}

message EigenVectorRequest {
  repeated RealRows rows = 1;
}

message EigenVectorResponse {
  repeated ComplexRows rows = 1;
}
